src/EditorFeatures/Core/Preview/DifferenceViewerPreview.cs

```diff
@@ -5,24 +5,39 @@
 #nullable disable
 
 using System;
+using Microsoft.CodeAnalysis.Editor.Shared.Preview;
 using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.VisualStudio.Text.Differencing;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
 {
-    internal class DifferenceViewerPreview : IDisposable
+    internal sealed class DifferenceViewerPreview : IDisposable
     {
         private IDifferenceViewer _viewer;
 
-        public DifferenceViewerPreview(IDifferenceViewer viewer)
+        private DifferenceViewerPreview(
+            IDifferenceViewer viewer,
+            ReferenceCountedDisposable<PreviewWorkspace> leftWorkspace,
+            ReferenceCountedDisposable<PreviewWorkspace> rightWorkspace)
         {
             Contract.ThrowIfNull(viewer);
             _viewer = viewer;
+            LeftWorkspace = leftWorkspace?.TryAddReference();
+            RightWorkspace = rightWorkspace?.TryAddReference();
```

Comment:
why `?.`?  You never are passed null instances in right?

Also, is this another location where we could use the asserting AddRef methods?

Reply:
Multiple locations pass in null

Reply:
➡️ These can be null. See 6a08a737958648f643d2aed618603b02c895e1c3